src/Nethermind/Nethermind.Serialization.Json/Abi/AbiParameterConverter.cs

```diff
@@ -0,0 +1,111 @@
+﻿//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Nethermind.Abi;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Nethermind.Serialization.Json.Abi
+{
+    internal static class AbiParameterConverterStatics
+    {
+        internal static readonly Regex TypeExpression = new Regex(@"^(?<T>u?int(?<M>\d{1,3})?|address|bool|u?fixed(?<P>(?<M>\d{1,3})x(?<N>\d{1,2}))?|bytes(?<M>\d{1,3})?|function|string|tuple)(?<A>\[(?<L>\d+)?\])?$",
```

Comment:
can we call the groups differently - M, P, L, T say nothing to me when I look at the code and they are used later in the code - maybe some

Reply:
Added comments I want to keep them short for readability